### ISSUE TITLE
make CFLAGS="-DNDEBUG" fails

### ISSUE THREAD

**Issue by tmakatos** [nutanix/libvfio-user#341]
```
make CFLAGS="-DNDEBUG"
...
[ 53%] Building C object samples/CMakeFiles/server.dir/server.c.o
/root/src/libvfio-user/samples/server.c: In function ‘migration_write_data’:
/root/src/libvfio-user/samples/server.c:361:9: error: variable ‘ret’ set but not used [-Werror=unused-but-set-variable]
     int ret;
```

**Comment by jlevon** [comment]
Not a bug, we intentionally make assert() work in release mode too

**Comment by tmakatos** [comment]
> Not a bug, we intentionally make assert() work in release mode too

I know, my understanding is that `CFLAGS` can be overridden and `-DNDEBUG` can be specified, breaking the build.

**Comment by jlevon** [comment]
Sure, but we don't need to support arbitrary cflags passed in. We decided to always have assert() on, so this is not a bug.

**Comment by tmakatos** [comment]
The problem is that the compilation fails, so we should either redefine `NDEBUG` or at least document that `-DNDEBUG` can't be specified in `CFLAGS`.

**Comment by jlevon** [comment]
Is there something that's expecting to do this?

In general we surely can't be expected to handle arbitrary compile options.

**Comment by tmakatos** [comment]
I don't know if there's currently a valid use case, I accidentally run into when SPDK builts libvfio-user as a submodule.

I think it's quite common for some compile options to be set, and `DNDEBUG` is one of them. If we rely on `DNEBUG`, then at least there should be a sanity check where we fail to compile if it's not set.

**Comment by jlevon** [comment]
fixed by #563
